tests/hlsl-intrinsic/helper-lane.slang

```diff
@@ -0,0 +1,10 @@
+//TEST:SIMPLE: -profile ps_6_0
+//TEST:SIMPLE: -profile ps_6_6
+
+float4 main() : SV_Target
+{
+    // IsHelperLane() has different implementations
+    // under 6.0 <= SM < 6.6, see the following:
+    // https://microsoft.github.io/DirectX-Specs/d3d/HLSL_ShaderModel6_6.html#is-helper-lane
+    return IsHelperLane() ? (float4) 0.0f : (float4) 1.0f;
+}
```

Comment:
We need to test whether or not the function call to `IsHelperLane()` is emitted to HLSL.
And we don't need to care how DXC converts it to actual DXIL.
The differentiation between 6.6 and the versions prior here is unnecessary.
The first two lines should merge and do the following,
```
//TEST:SIMPLE(filecheck=HLSL): -target hlsl -entry main -stage fragment

float4 main() : SV_Target
{
    //HLSL:IsHelperLane()
    return float(IsHelperLane());
}
```
With this change, you don't need the other "*.expected" files.
Please remove them.